fastify.js

```diff
@@ -917,6 +914,24 @@ function fastify (options) {
     this[kGenReqId] = reqIdGenFactory(this[kOptions].requestIdHeader, func)
     return this
   }
+
+  function acceptHTTPMethod (method, { hasBody = false } = {}) {
+    if (typeof method !== 'string' || http.METHODS.indexOf(method) === -1) {
+      throw new FST_ERR_ROUTE_METHOD_INVALID()
+    }
+
+    if (hasBody === true) {
+      this[kAcceptedHTTPMethods].bodywith.add(method)
+    } else {
+      this[kAcceptedHTTPMethods].bodyless.add(method)
+    }
```

Comment:
I would prevent some silly thing.
```js
fastify.acceptHTTPMethod('COPY')
fastify.acceptHTTPMethod('COPY', { hasBody: true })
```

```suggestion
    if (hasBody === true) {
      this[kAcceptedHTTPMethods].bodywith.add(method)
      this[kAcceptedHTTPMethods].bodyless.remove(method)
    } else {
      this[kAcceptedHTTPMethods].bodywith.remove(method)
      this[kAcceptedHTTPMethods].bodyless.add(method)
    }
```